src/librustc_resolve/late/diagnostics.rs

```diff
@@ -484,10 +503,25 @@ impl<'a> LateResolutionVisitor<'a, '_, '_> {
             let (followed_by_brace, closing_brace) = self.followed_by_brace(span);
             let mut suggested = false;
             match source {
-                PathSource::Expr(Some(parent)) => {
+                PathSource::Expr(Some(
+                    parent @ Expr { kind: ExprKind::Field(..) | ExprKind::MethodCall(..), .. },
```

Comment:
I think I've seen this in a different PR.
(I'll wait for rebase.)

Reply:
Coincidentally was just doing so.